annotated_test.go

```diff
@@ -1474,6 +1474,159 @@ func TestAnnotate(t *testing.T) {
 	})
 }
 
+func TestAnnotateApply(t *testing.T) {
+	t.Parallel()
+
+	type a struct{}
+	type b struct{ a *a }
+	newA := func() *a { return &a{} }
+	newB := func(a *a) *b {
+		return &b{a}
+	}
+
+	var (
+		errTagSyntaxSpace            = ` multiple tags are not separated by space`
+		errTagKeySyntax              = " tag key is invalid, Use group, name or optional as tag keys"
+		errTagValueSyntaxQuote       = `tag value should start with double quote. Did you mean `
+		errTagValueSyntaxEndingQuote = `tag value should end in double quote. Did you mean `
+		errTagSyntaxColon            = `tag key and value not separated by colon. Did you mean `
+		errTagValueSpace             = `tag key value separated by extra space. Use format` + " `tag_key:\"value\"`"
+	)
+	tests := []struct {
+		give           string
+		wantErr        string
+		giveAnnotation fx.Annotation
+	}{
+		{
+			give:           "ParamTags wrong ending quote",
+			wantErr:        errTagValueSyntaxEndingQuote + "`name:\"something\"` ?",
```

Comment:
Considering specifying in the test message whether it is the key or the value that has the incorrect quote